harvester/db_api_functions.py

```diff
@@ -0,0 +1,94 @@
+    :param harvest_url: endpoint for harvesting
+    :param timeout: Request timeout in seconds
+    :return: JSON response (dict) containing 'harvest_run_id', optionally 'last_harvest_date', and endpoint config; returns None on error.
+    """
+    url = f"{API_BASE_URL}/harvest_run"
+    payload = {"harvest_url": harvest_url}
+    try:
+        response = requests.post(url, json=payload, timeout=timeout)
+        response.raise_for_status()
+        run_info = response.json()
+        logger.info("Started harvest run id=%s.", run_info.get("id"))
+        return run_info
+    except requests.RequestException as e:
+        logger.error("Failed to start harvest run for %s: %s", harvest_url, e)
+        return None
+
+def get_open_run_id(harvest_url: str, timeout: int = 30) -> Optional[Dict]:
+    """
+    GET /harvest_run to fetch an open harvest run ID if it exists.
+
+    :param harvest_url: endpoint for harvesting
+    :param timeout: Request timeout in seconds
+    :return: JSON response (dict) containing the ID of a harvest run and its status, or None if not found or failed
+    """
+    url = f"{API_BASE_URL}/harvest_run"
+    params = {"harvest_url": harvest_url}
+    try:
+        response = requests.get(url, params=params, timeout=timeout)
+        response.raise_for_status()
+
+        response = response.json()
+        if response["status"] == "open":
+            return response["id"]
+        else:
+            return None
+
+    except requests.RequestException as e:
+        logger.error("Error checking for open harvest run for %s: %s", harvest_url, e)
+        return None
+
+def close_harvest_run(api_base_url: str, payload: Dict) -> None:
+    """
+    PUT /harvest_run to close the harvest run.
+
+    :param api_base_url: API endpoint
+    :param payload: payload for API post request to close the harvest run
+    """
+    url = f"{api_base_url}/harvest_run"
+    run_id = payload.get("id")
+    try:
+        response = requests.put(url, json=payload, timeout=30)
+        response.raise_for_status()
+        logger.info(
+            "Closed harvest run %s — started %s, finished %s",
+            run_id,
+            payload.get("started_at"),
+            payload.get("completed_at"),
+        )
+    except requests.RequestException as e:
+        if e.response is not None:
+            logger.error("Failed to close harvest run %s: API error %s %s", run_id, e.response.status_code, e.response.text)
+        else:
+            logger.error("Failed to close harvest run %s: %s", run_id, e)
+
+
+def send_harvest_event(api_base_url: str, event_payload: Dict) -> bool:
+    """
+    Send event_payload to API.
+
+    :param api_base_url: API endpoint
+    :param event_payload: dictionary containing event data for harvest_event route
+    :return logical: True if the payload has been sent to API successfully 
+    """
+    url = f"{api_base_url}/harvest_event"
+    try:
+        response = requests.post(url, json=event_payload, timeout=60)
```

Comment:
Every call does a fresh TCP + TLS handshake.
I think it will be good to have only one shared session and reuse it 

`_SESSION = requests.Session()`

then everywhere use
`response = _SESSION.post(url, json=event_payload, timeout=timeout)`

Reply:
Do you do now do sync or async requests?

Reply:
That is fine as it won't create high load